umpost/um2netcdf.py

```diff
@@ -99,46 +126,193 @@ def convert_proleptic(time):
+    else:
+        lon_coordinate.var_name = VAR_NAME_LON_STANDARD
+
+
+def is_lat_river_grid(latitude_points):
+    """
+    Check whether latitude points are on the river routing grid.
+
+    Parameters
+    ----------
+    latitude_points: (array) 1D array of latitude grid points.
+    """
+    return len(latitude_points) == NUM_LAT_RIVER_GRID_POINTS
+
+
+def is_lon_river_grid(longitude_points):
+    """
+    Check whether longitude points are on the river routing grid.
+
+    Parameters
+    ----------
+    longitude_points: (array) 1D array of longitude grid points.
+    """
+
+    return len(longitude_points) == NUM_LON_RIVER_GRID_POINTS
+
+
+def is_lat_v_grid(latitude_points, grid_type, dlat):
+    """
+    Check whether latitude points are on the lat_v grid.
+
+    Parameters
+    ----------
+    latitude_points: (array) 1D array of latitude grid points.
+    grid_type: (string) model horizontal grid type.
+    dlat: (float) meridional spacing between latitude grid points.
+    """
+    min_latitude = latitude_points[0]
+    min_lat_v_nd_grid = -90.+0.5*dlat
+    min_lat_v_eg_grid = -90
+
+    if grid_type == GRID_END_GAME:
+        return min_latitude == min_lat_v_eg_grid
+    elif grid_type == GRID_NEW_DYNAMICS:
+        return np.allclose(min_lat_v_nd_grid, min_latitude)
     else:
-        lat.var_name = 'lat'
-
-    lon = cube.coord('longitude')
-    if len(lon.points) == 360:
-        lon.var_name = 'lon_river'
-    elif (lon.points[0] == 0 and grid_type == 'EG') or \
-         (np.allclose(0.5*dlon, lon.points[0]) and grid_type == 'ND'):
-        lon.var_name = 'lon_u'
+        return False  # TODO: Is this situation ever valid?
+
+
+def is_lon_u_grid(longitude_points, grid_type, dlon):
+    """
+    Check whether longitude points are on the lon_u grid.
+
+    Parameters
+    ----------
+    longitude_points: (array) 1D array of longitude grid points.
+    grid_type: (string) model horizontal grid type.
+    dlon: (float) zonal spacing between longitude grid points.
+    """
+    min_longitude = longitude_points[0]
+    min_lon_u_nd_grid = 0.5*dlon
+    min_lon_u_eg_grid = 0
+
+    if grid_type == GRID_END_GAME:
+        return min_longitude == min_lon_u_eg_grid
+    elif grid_type == GRID_NEW_DYNAMICS:
+        return np.allclose(min_lon_u_nd_grid, min_longitude)
     else:
-        lon.var_name = 'lon'
+        return False  # TODO: Is this situation ever valid?
+
+
+def add_latlon_coord_bounds(cube_coordinate):
+    """
+    Add bounds to horizontal coordinate (longitude or latitude) if
+    they don't already exist. Edits coordinate in place.
+
+    Parameters
+    ----------
+    cube_coordinate: coordinate object from iris cube.
+    """
+    coordinate_name = cube_coordinate.name()
+    if coordinate_name not in [LONGITUDE, LATITUDE]:
+        raise ValueError(
+                f"Wrong coordinate {coordinate_name} supplied. "
+                f"Expected one of {LONGITUDE}, {LATITUDE}."
+            )
```

Comment:
Are we ever likely to get coordinate names other than `LATITUDE` and `LONGITUDE`? If so, it might be worth parameterizing out the 'set of coordinate names I expect' for use in this situation. You could also do it using `GLOBAL_COORDS_BOUNDS.keys()`.

Reply:
I added in this error to try to guard against `add_latlon_coord_bounds` being inadvertently used on the wrong coordinates, and so here we should only be expecting `LATITUDE` or `Longitude`.

Would adding a constant eg `HORIZONTAL_COORD_NAMES = [LATITUDE, LONGITUDE]`, and then checking `if coordinate_name not in HORIZONTAL_COORD_NAMES` be a bit cleaner here?

Reply:
Precisely, although you've possibly already got the 'constant' available as `GLOBAL_COORDS_BOUNDS.keys()`.

Reply:
Is `GLOBAL_COORDS_BOUNDS` likely to change, therefore adding keys that `coord name` should not be?

The advantage of the current line is that the test condition is obvious.

Reply:
Also a minor formatting thing, lines 266-268 look over indented.

Reply:
Have updated the indentation in 37049c7